exercises/diffie-hellman/.meta/template.j2

```diff
@@ -0,0 +1,35 @@
+{%- import "generator_macros.j2" as macros with context -%}
+{% set class = exercise | camel_case -%}
+{{ macros.header(["private_key", "public_key", "secret"]) }}
+
+class {{ exercise | camel_case }}Test(unittest.TestCase):
+    {% for case in cases -%}
+    {% set property = case["property"] | to_snake -%}
+    {% set description = case["description"] | to_snake -%}
+    {% set expected = case['expected'] %}
+    {% set input = case['input'] %}
+    {% if property == "private_key_is_in_range" -%}
+    def test_{{ description }}(self):
+        primes = [5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47]
+        for i in primes:
+            self.assertTrue(1 < private_key(i) < i)
+    {% elif property == "private_key_is_random" -%}
+    def test_{{ description }}(self):
+        p = 2147483647
+        private_keys = []
+        for i in range(5):
+            private_keys.append(private_key(p))
+        self.assertEqual(len(set(private_keys)), len(private_keys))
+    {% elif property == "key_exchange"%}
+    def test_{{ description }}(self):
+        {% for key, val in input.items() -%}
+            {{ key | to_snake }} = {{ val  | string | to_snake_base }}
```

Comment:
Can you explain why `to_snake` was inadequate here?

Reply:
In this case most of the input most values are function calls, therefore old `to_snake` function would replace punctuation (brackets and commas in this case) with underscore. 

I've reveted `to_snake_base` back to one function but now `to_snake` takes optional that @yawpitch suggested.